apps/desktop/src/renderer/components/Chat/ChatInterface/components/TiptapPromptEditor/parseTextToEditorContent.ts

```
@@ -0,0 +1,46 @@
+export function parseTextToEditorContent(text: string): JSONContent {
+	const paragraphs = text.split("\n").map((line): JSONContent => {
+		if (line === "") {
+			return { type: "paragraph" };
+		}
+
+		const inlineNodes: JSONContent[] = [];
+		let lastIndex = 0;
+		MENTION_RE.lastIndex = 0;
+
+		let match: RegExpExecArray | null;
+		while ((match = MENTION_RE.exec(line)) !== null) {
```

Biome (2.4.10)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)